core/domain/topic_fetchers.py

```diff
@@ -396,7 +446,9 @@ def get_topic_summary_from_model(topic_summary_model):
     )
 
 
-def get_topic_summary_by_id(topic_id, strict=True):
+def get_topic_summary_by_id(
+    topic_id: str, strict: bool = True
+) -> Optional[topic_domain.TopicSummary]:
```

Comment:
Again, can this be overloaded?

Reply:
Not sure really but according to the use cases, it seems like it is not needed.